internal/framework/generic_resource.go

```diff
@@ -306,8 +307,8 @@ func (r *GenericResource) GetTimeout(operation string, defaultTimeout time.Durat
 
 func (r *GenericResource) AwaitReady(ctx context.Context, state ResourceState, operation string) error {
 	timeout := r.GetTimeout(operation, READINESS_TIMEOUT)
-	if timeout <= 0 {
-		// TODO(asz6): Log message saying that we are not waiting
+	if timeout == 0 {
```

Comment:
What about negative duration?

Reply:
We validate that the timeouts are non-negative in line 272.